### PR TITLE
Mes 7482

### PR DESCRIPTION
**Pull request by turnerc1**
## Description

A Successful Cat-B test

## Checklist

- [ ] PR title includes the JIRA ticket number and an understandable description
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket

### CODE REVIEW

File: src/pageobjects/des/finalOutcomeScreen.pageobject.ts [hunk 1]
```diff
@@ -9,6 +39,72 @@ class FinalOutcomePageObject extends Page {
...
+      await setInputField('add', passCertNumber, this.passCertificateNumberInput);
+    }
+
+    await scroll(this.d255YesInput);
+
+    if (d255.toLowerCase() !== 'na') {
+      if (d255.toLowerCase() === 'yes') {
+        await clickElement('click', 'selector', this.d255YesInput);
+      }
+      if (d255.toLowerCase() === 'no') {
```
Comment (dhDVSA) [inline src/pageobjects/des/finalOutcomeScreen.pageobject.ts:79]: just a slight suggestion :

could do a switch statement here 
switch(d255.toLowerCase())
case yes
 await clickElement('click', 'selector', this.d255YesInput);
case no
 await clickElement('click', 'selector', this.d255NoInput);

Why is the condition for d255 to no be 'na'

**Review by dhDVSA — APPROVED** [review]

File: src/pageobjects/des/finalOutcomeScreen.pageobject.ts [hunk 2]
```diff
@@ -9,6 +39,72 @@ class FinalOutcomePageObject extends Page {
...
+  async completePassedFinalOutcomePage(data: any): Promise<void> {
+    const {
+      provisionalLicenceReceived, transmission, passCertNumber,
+      d255, testLanguage, debriefWitnessed,
+    } = data;
+
+    await checkEqualsText('element', this.testOutcomePassed, false, 'Passed');
+
+    if (provisionalLicenceReceived.toLowerCase() !== 'na') {
+      if (provisionalLicenceReceived.toLowerCase() === 'yes') {
```
Comment (Cullima) [inline src/pageobjects/des/finalOutcomeScreen.pageobject.ts:52]: why do we need the outer `if`?

File: src/pageobjects/des/finalOutcomeScreen.pageobject.ts [hunk 3]
```diff
@@ -9,6 +39,72 @@ class FinalOutcomePageObject extends Page {
...
+      }
+    }
+
+    if (passCertNumber.toLowerCase() !== 'na') {
+      await setInputField('add', passCertNumber, this.passCertificateNumberInput);
+    }
+
+    await scroll(this.d255YesInput);
+
+    if (d255.toLowerCase() !== 'na') {
```
Comment (Cullima) [inline src/pageobjects/des/finalOutcomeScreen.pageobject.ts:75]: same comment here

File: src/pageobjects/des/office.pageobject.ts [hunk 4]
```diff
@@ -1,7 +1,91 @@
...
+
+  get weatherConditionsSelector() { return ('des-office-page::weather-conditions-selector'); }
+
+  async completeOfficePage(data: any): Promise<void> {
+    const {
+      activityCode, routeNumber, independentDriving, trueLikeness, distinguishingFeatures,
+      identification, showMeQuestion, weatherCondition,
+    } = data;
+
+    if (activityCode.toLowerCase() !== 'na') {
```
Comment (Cullima) [inline src/pageobjects/des/office.pageobject.ts:34]: mark as `@TODO` if further work required

File: src/pageobjects/des/office.pageobject.ts [hunk 5]
```diff
@@ -1,7 +1,91 @@
...
+      // Code will need to be inputted for Unsuccesful and Terminated tests
+    }
+
+    await setInputField('add', routeNumber, this.routeNumberInput);
+
+    if (independentDriving.toLowerCase() !== 'na') {
```
Comment (Cullima) [inline src/pageobjects/des/office.pageobject.ts:40]: same comment with these outer `if`s

**Review by Cullima — APPROVED** [review]

File: src/pageobjects/des/finalOutcomeScreen.pageobject.ts [hunk 6]
```diff
@@ -9,6 +39,72 @@ class FinalOutcomePageObject extends Page {
...
+    }
+
+    if (transmission.toLowerCase() !== 'na') {
+      if (transmission.toLowerCase() === 'manual') {
+        await clickElement('click', 'selector', this.transmissionManualInput);
+      }
+      if (transmission.toLowerCase() === 'automatic') {
+        await clickElement('click', 'selector', this.transmissionAutomaticInput);
+      }
+    }
```
Comment (matthew2564) [inline src/pageobjects/des/finalOutcomeScreen.pageobject.ts:67]: If you wanted to remove the 3 if statements into one, you could do something like this

```
    const transmissionValue = transmission?.toLowerCase();

    if (transmissionValue !== 'na') {
      await clickElement(
        'click',
        'selector',
        transmissionValue === 'manual' ? this.transmissionManualInput : this.transmissionAutomaticInput,
      );
    }
```

**Review by Cullima — CHANGES_REQUESTED** [review]

**Review by Cullima — APPROVED** [review]

**Pull request merged**